src/api/mod.rs

```diff
@@ -115,3 +137,154 @@ fn build_fetch_query<T: Serialize>(path: &str, form: T) -> String {
   let form_str = serde_urlencoded::to_string(&form).unwrap_or(path.to_string());
   format!("{path}?{form_str}")
 }
+
+#[cfg(not(feature = "ssr"))]
```

Comment:
I don't think we have to worry about cookies on the browser side.

Reply:
this was created for completeness as i was developing the whole solution. 

i thought it could be useful in the future and also the code implied that there should be an equivalent function when compiling in "ssr" and "hydrate" modes.

maybe we can refactor this away later as part of a cookie oriented refactoring?